gcpy/file_regrid.py

```diff
@@ -448,157 +551,457 @@ def regrid_ll_to_cssg(
+            if "lat" not in dset[var].dims       \
+                and "lon" not in dset[var].dims
             ]
-        ds_in = ds_in.drop(["lat_bnds", "lon_bnds"])
-        non_fields_ds = ds_in[non_fields]
-        ds_in = ds_in.drop(non_fields)
+        dset = dset.drop(["lat_bnds", "lon_bnds"])
+        non_fields = dset[non_fields]
+        dset = dset.drop(non_fields)
 
-        # Create the regridder
+        # Create the regridder and regrid the data
         regridder = make_regridder_L2L(
         ll_res_in,
             ll_res_out,
             reuse_weights=True,
             in_extent=in_extent,
-            out_extent=out_extent
+            out_extent=out_extent,
+            weightsdir=weightsdir
+        )
+        dset = regridder(
+            dset,
+            keep_attrs=True
         )
-        ds_out = regridder(ds_in, keep_attrs=True, verbose=False)
 
         # Add the non-regriddable fields back
-        ds_out = ds_out.merge(non_fields_ds)
+        dset = dset.merge(non_fields)
 
         # Change order of dimensions
-        ds_out = ds_out.transpose(
-            'time', 'lev', 'ilev', 'lat', 'lon', ...
+        dset = dset.transpose(
+            "time", "lev", "ilev", "lat", "lon", ...
+        )
+
+        # Set the lev:positive attribute accordingly
+        dset = flip_lev_coord_if_necessary(
+            dset,
+            dim_format_in="classic",
+            dim_format_out="classic"
+        )
+
+        # Save lat/lon coordinate metadata
+        dset = save_ll_metadata(
+            dset,
+            verbose=verbose
         )
 
-    return ds_out
+    return dset
+
+
+def flip_lev_coord_if_necessary(
+        dset,
+        dim_format_in,
+        dim_format_out
+):
+    """
+    Determines whether it is necessary to flip the "lev" and "ilev"
+    coords of an xarray.Dataset in the vertical depending on the
+    values ofdim_format_in and dim_format_out.  Also sets the
+    attributes "lev:positive" and "ilev:positive" accordingly.
+
+    Args:
+    -----
+    dset : xarray.Dataset
+        The input dataset.
+    dim_format_in : str
+        Input grid format ("classic", "checkpoint", "diagnostic").
+    dim_format_out : str
+        Output grid format ("classic", "checkpoint", "diagnostic").
+
+    Args:
+    -----
+    dset : xarray.Dataset
+        The modified dataset.
+
+    Remarks:
+    --------
+    (1) classic    : lev is in ascending order  (lev:positive="up"  )
+    (2) diagnostic : lev is in ascending order* (lev:positive="up"  )
+    (3) checkpoint : lev is in descending order (lev:positive="down")
+
+    *Except for the Emissions collection, which has lev arranged
+     in descending order.
+    """
+    verify_variable_type(dset, xr.Dataset)
+    verify_variable_type(dim_format_in, str)
+    verify_variable_type(dim_format_out, str)
+
+    # ==================================================================
+    # checkpoint/diagnostic to classic: lev, ilev in ascending order
```

Comment:
Could start each of these headers with Case 1, Case 2, Case3, etc to make it clear it is going through all of the input/output combinations.

Reply:
Now updated in commit 922f233.